### DISCUSSION TITLE
Feature Request - Can wizard attributes for nextLabel and backLabel evaluated at every step

### DISCUSSION THREAD

**Discussion by mores** [primefaces/community#2216] [PrimeFaces] [answered]
Currently in version 14.0.1 it looks like nextLabel and backLabel on wizard is only evaluated once.
Can an update be made so button labels are updated on every step ?

**Comment by melloware** [comment]
cc @FlipWarthog i feel like this has come up before?  Maybe Stack overflow I think someone had a JS hack to allow you to change it per step?

**Reply by melloware** [reply, marked as answer]
@mores here it is in the docs: https://primefaces.github.io/primefaces/14_0_0/#/components/wizard?id=custom-ui

**Reply by mores** [reply]
Thats great !
Perhaps the example could be flushed out a little more like
```
<p:outputPanel id="navigationOutputPanel">
    <p:commandButton value="Back" icon="pi pi-arrow-left" onclick="PF('wiz').back();" rendered="#{userWizard.displayBackButton}" update="navigationOutputPanel" />
   <p:commandButton value="#{userWizard.nextLabel}" icon="pi pi-arrow-right" onclick="PF('wiz').next();" iconPos="right" update="navigationOutputPanel" />
</p:outputPanel>

**Reply by melloware** [reply]
yes!

**Reply by melloware** [reply]
Docs updated: https://github.com/primefaces/primefaces/pull/12208

**Reply by mores** [reply]
Great !
Thank you.
